### ISSUE TITLE
`StringIndexOutOfBoundsException`

### ISSUE THREAD

**Issue by cddf** [stefan-niedermann/nextcloud-notes#1435]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**
The notes app contains some unsynchronized notes and crashes on each start. Deleting the cache did not help and I don't want to reset the app as long as there is hope to get some important notes back.  
(It is not my phone and therefore my testing capabilities are limited.)

I see, transferring the app between phones is somehow a corner case producing weird states, but the option is likely to be used. I hope, the stacktrace points to an obvious issue, which is easy to solve.

Thank you for providing the app :)

**To Reproduce**
Here's the history prior to the bug:
1. The app was used on a Samsung A50.
2. It was transferred to the new phone using a preinstalled app by Samsung.
3. The login to the Nextcloud app was lost during the transfer.
4. The notes app worked some time without synchronizing.
5. Suddenly the app crashes and does not work anymore.
6. Reconnecting the Nextcloud app to the server did not help to avoid the crash.

**Expected behavior**
The app should start without crashing.

**Smartphone:**
```
App Version: 3.4.16
App Version Code: 3004016
App Flavor: play

Files App Version Code: 30180190

OS Version: 4.14.113-22340613(N975FXXS7FUJ1)
OS API Level: 30
Device: d2s
Manufacturer: samsung
Model (and Product): SM-N975F (d2seea)
```

**Server**
 - Nextcloud version: 22.2.3
 - Nextcloud Notes version (server app): 4.2.0

**Stacktrace**

```java
java.lang.StringIndexOutOfBoundsException: length=0; index=1
	at java.lang.String.substring(String.java:2060)
	at it.niedermann.owncloud.notes.main.slots.SlotterUtil.fillListByInitials(SlotterUtil.java:61)
	at it.niedermann.owncloud.notes.main.MainViewModel.fromNotes(MainViewModel.java:289)
	at it.niedermann.owncloud.notes.main.MainViewModel.lambda$getNotesListLiveData$3$it-niedermann-owncloud-notes-main-MainViewModel(MainViewModel.java:268)
	at it.niedermann.owncloud.notes.main.MainViewModel$$ExternalSyntheticLambda5.apply(Unknown Source:8)
	at androidx.lifecycle.Transformations$1.onChanged(Transformations.java:76)
	at androidx.lifecycle.MediatorLiveData$Source.onChanged(MediatorLiveData.java:152)
	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
	at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:246)
	at android.app.ActivityThread.main(ActivityThread.java:8633)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

**Comment by stefan-niedermann** [comment]
Uhm, let me have a look at this bug, i didn't encounter that yet. I guess we have good chances to provide a bug fix release qhich should allow you access to your notes again, so you can back them up before removing and reimporting your account for a clean state.... could you please send me a mail with aa reference to this issue to info@niedermann.it ?

**Comment by stefan-niedermann** [comment]
Looks like you managed to have a note with a completely blank title which could lead to the broken state when the list is sorted alphabetically. I added a check for this situation which will be released with `3.4.17`. It might take some time until January until this versions hits Play Store in the end, therefore just contact me, so I can send you the new version immediately :)
